### PR TITLE
[fix][io] use create for bulk op if target cluster is ElasticSearch

### PR DESCRIPTION
**Pull request by zzzz465**
### Motivation

on ElasticSearch, the bulk API must be called with `create` operation if the target index is data stream.
the current plugin cannot insert documents to elasticsearch data stream if the bulk is enabled, throwing `illegal_argument_exception`

this PR address this issue.

### Modifications

- adds `INDEX_TYPE` config
- use `CREATE` op if `INDEX_TYPE` is `DATA_STREAM`

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @zzzming @nicoloboschi can you give a look at this patch?

**Comment by zzzz465** [comment]
@tisonkun @zzzming @nicoloboschi Hi, can you review this PR?

**Comment by shibd** [comment]
@zzzz465 I will review this. Can you resolve conflicts first?

File: pulsar-io/elastic-search/pom.xml [hunk 1]
```diff
@@ -68,6 +68,12 @@
       <artifactId>jackson-databind</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-core</artifactId>
+      <version>${jackson.version}</version>
```
Comment (nicoloboschi) [inline pulsar-io/elastic-search/pom.xml:74]: no need for the version

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
good work! can you add a test for both OS 1.x and OS 2.x with the data-stream case?

**Comment by zzzz465** [comment]
I'm resolving conflicts, I accidently closed the PR. can you re-open it?

**Comment by zzzz465** [comment]
> @zzzz465 The button told that "There is no new commits".
> 
> I suggest you simply create a new PR and try to use a different branch name, using "master" identical to the upstream can cause a few subtle issues on collaboration.

I've created new PR here https://github.com/apache/pulsar/pull/21377

**Pull request closed without merging**